<launch>

    <master auto="start"/>

    <!-- Arguments -->
    <arg name="move_forward_only"   default="false"/>
    <arg name="odom_frame_id"       default="odom"/>
    <arg name="base_frame_id"       default="base_link"/>
    <arg name="global_frame_id"     default="map"/>
    <arg name="cmd_vel_topic"       default="/navi/cmd_vel" />
    <arg name="odom_topic"          default="/base_controller/odom" />
    <arg name="laser_topic"         default="/scan_filtered"/>
    <arg name="map_topic"           default="/map" />

    <!-- Remapping Topics -->
    <remap from="map" to="$(arg map_topic)" />

    <!-- Move base node -->
    <node name="move_base_node" pkg="move_base" type="move_base" respawn="false" output="screen">
        <rosparam file="$(find logcart_navigation)/param/move_base_params.yaml" command="load" />

        <rosparam file="$(find logcart_navigation)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find logcart_navigation)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />

        <rosparam file="$(find logcart_navigation)/param/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find logcart_navigation)/param/global_costmap_params.yaml" command="load" />

        <rosparam file="$(find logcart_navigation)/param/dwa_local_planner_params.yaml" command="load" />
        <rosparam file="$(find logcart_navigation)/param/global_planner_params.yaml" command="load" />
        <rosparam file="$(find logcart_navigation)/param/navfn_global_planner_params.yaml" command="load" />

        <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
        <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
        <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
        <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
        <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>
    </node>

</launch>
